src/decoder.rs

```diff
@@ -103,9 +114,12 @@ fn decode_signature_struct(
         _ => return Err(CoseError::UnexpectedType),
     };
 
-    let ee_cert = &get_map_value(&protected_signature_header, &CborType::Integer(4))?;
+    let ee_cert = &get_map_value(protected_signature_header, &CborType::Integer(4))?;
     let ee_cert = unpack!(Bytes, ee_cert).clone();
 
+    // The unprotected header section is expected to be an empty map.
+    validate_empty_map(&cose_signature[1])?;
```

Comment:
Maybe `is_empty_map`? Not sure what I expect `validate_` to do but not necessarily to check whether it's an empty map or not.

Reply:
Ok - I changed the name and added some documentation.